bundle/client/grpc/main.go

```diff
@@ -66,10 +68,50 @@ func decodePayload(resp *pbGW.Content) {
 	*/
 }
 
+func getServiceToken(authAddr string, tls *tls.Config) (string, error) {
+	reqBody := map[string]string{
+		"grant_type":    string(service.AllowedGrantType_CLIENT_CREDENTIALS),
+		uri.ClientIDKey: service.ClientTest,
+		uri.AudienceKey: "test",
+	}
+	d, err := json.Encode(reqBody)
+	if err != nil {
+		return "", err
+	}
+
+	request, err := http.NewRequest(http.MethodPost, "https://"+authAddr+"/oauth/token", bytes.NewReader(d))
+	if err != nil {
+		return "", err
+	}
+	c := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tls,
+		},
+	}
+	res, err := c.Do(request)
+	if err != nil {
+		return "", err
+	}
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("returns statu code %v", res.StatusCode)
+	}
+	var body map[string]string
+	err = json.ReadFrom(res.Body, &body)
+	if err != nil {
+		return "", err
+	}
+	token := body["access_token"]
+	if token == "" {
+		return "", fmt.Errorf("token not found in body")
+	}
+	return token, nil
+}
+
 func main() {
-	addr := flag.String("addr", "localhost:9084", "address")
+	addr := flag.String("addr", "localhost:8443", "address")
```

Comment:
address localhost:8443 is configurable and reserved by the nginx; not ? or even if you set FQDN to something else this will work - because it;s inside of the container? Won't you have problems with CORS?

Reply:
grpc is not a part of bundle - you need to compile it separetly

Reply:
CORS is part of browser it is not check by servers.